tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -257,7 +267,14 @@ let main argv = 0"""
                     |> List.map (fun cmpl ->
                             match cmpl with
                             | CompilationReference (cmpl, staticLink) ->
-                                compileCompilationAux disposals ignoreWarnings cmpl, staticLink)
+                                compileCompilationAux disposals ignoreWarnings cmpl, staticLink
+                            | TestCompilationReference (cmpl) -> 
+                                let tmp = Path.GetTempFileName()
+                                disposals.Add({ new IDisposable with 
+                                                    member _.Dispose() = 
+                                                        try File.Delete tmp with | _ -> () })
+                                cmpl.EmitAsFile tmp
+                                (([||], tmp), []), false)
```

Comment:
I take it this was just leaving stuff around?

Reply:
This is all new, so nothing was being left around before. But yes, once I create a tmp file, I need to make sure we always try to delete it on dispose.